src/app/core/metric-module/metric.service.ts

```diff
@@ -74,6 +75,14 @@ export class MetricService {
     };
   }
 
+  async getTagMetrics(tagId?: string): Promise<TagMetrics> {
+    const tagMetrics: TagMetrics = await this.http
+      .get<TagMetrics>(METRIC_ROUTES.GET_TAG_METRICS(tagId))
+      .pipe(catchError(this.handleError))
+      .toPromise();
+    return tagMetrics as TagMetrics;
+  }
+
```

Comment:
Exactly like you did this here is what we can do for the searchLearningObjects and the getTags requests I left comments on above! Those services already have the implementation but this is the same idea of having these requests in the service files, then calling that in the component files